sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadAsyncClient.java

```diff
@@ -57,21 +87,21 @@ public void sendChatReadReceipt(String chatThreadId, SendReadReceiptRequest send
 
     /**
      * Sends a message to a thread.
-     *
+     * 
      * @param chatThreadId The thread id to send the message to.
      * @param sendChatMessageRequest Details of the message to send.
      * @param callback the Callback that receives the response.
      * @throws IllegalArgumentException thrown if parameters fail the validation.
      * @throws ErrorException thrown if the request is rejected by server.
      * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
      */
-    public void sendChatMessage(String chatThreadId, SendChatMessageRequest sendChatMessageRequest, final Callback<String> callback) {
+    public void sendChatMessage(String chatThreadId, SendChatMessageRequest sendChatMessageRequest, final Callback<SendChatMessageResult> callback) {
```

Comment:
As discussed, we want this callback return type to be String, which is the message id of the SendChatMessageResult. Not sure if you unpack the callback here to String return type without touching Implement file.

Reply:
I made some manual change to convenience layer for that purpose. Please take a look.